src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs

```diff
@@ -160,87 +178,89 @@ public override bool Equals(object o)
 
             public static bool operator ==(RelationshipEntry re1, RelationshipEntry re2)
             {
-                object owner1 = (re1.Owner.IsAlive ? re1.Owner.Target : null);
-                object owner2 = (re2.Owner.IsAlive ? re2.Owner.Target : null);
-                return owner1 == owner2 && re1.Member.Equals(re2.Member);
+                object owner1 = (re1._owner.IsAlive ? re1._owner.Target : null);
+                object owner2 = (re2._owner.IsAlive ? re2._owner.Target : null);
+                return owner1 == owner2 && re1._member.Equals(re2._member);
             }
 
             public static bool operator !=(RelationshipEntry re1, RelationshipEntry re2)
             {
                 return !(re1 == re2);
             }
 
-            public override int GetHashCode()
-            {
-                return _hashCode;
-            }
+            public override int GetHashCode() => _hashCode;
         }
     }
 
     /// <summary>
-    ///    This class represents a single relationship between an object and a member.
+    /// This class represents a single relationship between an object and a member.
     /// </summary>
     public readonly struct MemberRelationship
     {
+        private readonly object _owner;
+        private readonly MemberDescriptor _member;
```

Comment:
Why did you explicitly declare these rather than letting the same fields be implicitly declared as part of the get-only auto props?

Reply:
Fixed

Reply:
Because the layout of the struct changes after converting them to auto props.

It used to be `owner`, then `member`, but became `member` then `owner`.

This breaks the netfx tests because there is a test that checks:

```cs
[Fact]
public void Indexer_NullSourceOwner_ThrowsArgumentNullException()
{
    MemberDescriptor member = TypeDescriptor.GetProperties(typeof(TestClass))[0];

    var service = new TestMemberRelationshipService();
    Assert.Throws<ArgumentNullException>("sourceOwner", () => service[null, member]);
    Assert.Throws<ArgumentNullException>("sourceOwner", () => service[null, member] = new MemberRelationship());

    Assert.Throws<ArgumentNullException>("Owner", () => service[new FakeMemberRelationsip { Owner = null, Member = member }.ToActual()]);
    Assert.Throws<ArgumentNullException>("Owner", () => service[new FakeMemberRelationsip { Owner = null, Member = member }.ToActual()] = new MemberRelationship());
}

[Fact]
public void Indexer_NullSourceMember_ThrowsArgumentNullException()
{
    var service = new TestMemberRelationshipService();
    Assert.Throws<ArgumentNullException>("sourceMember", () => service[new object(), null]);
    Assert.Throws<ArgumentNullException>("sourceMember", () => service[new object(), null] = new MemberRelationship());

    Assert.Throws<ArgumentNullException>("Member", () => service[new FakeMemberRelationsip { Owner = new object(), Member = null }.ToActual()]);
    Assert.Throws<ArgumentNullException>("Member", () => service[new FakeMemberRelationsip { Owner = new object(), Member = null }.ToActual()] = new MemberRelationship());
}

private struct FakeMemberRelationsip
{
    public object Owner;
    public MemberDescriptor Member;

    public MemberRelationship ToActual()
    {
        MemberRelationshipWrapper wrapper = default;
        wrapper.Fake = this;
        return wrapper.Actual;
    }
}

[StructLayout(LayoutKind.Explicit)]
private struct MemberRelationshipWrapper
{
    [FieldOffset(0)] public MemberRelationship Actual;
    [FieldOffset(0)] public FakeMemberRelationsip Fake;
}
```

The reason I am using this explicit layout hack is because the indexer:
```cs
/// <summary>
/// Returns the current relationship associated with the source, or MemberRelationship.Empty if
/// there is no relationship. Also sets a relationship between two objects. Empty
/// can also be passed as the property value, in which case the relationship will
/// be cleared.
/// </summary>
[SuppressMessage("Microsoft.Design", "CA1043:UseIntegralOrStringArgumentForIndexers")]
public MemberRelationship this[MemberRelationship source]
{
    get
    {
        if (source.Owner == null)
        {
            throw new ArgumentNullException(nameof(MemberRelationship.Owner));
        }
        if (source.Member == null)
        {
            throw new ArgumentNullException(nameof(MemberRelationship.Member));
        }

        return GetRelationship(source);
    }
    set
    {
        if (source.Owner == null)
        {
            throw new ArgumentNullException(nameof(MemberRelationship.Owner));
        }
        if (source.Member == null)
        {
            throw new ArgumentNullException(nameof(MemberRelationship.Member));
        }

        SetRelationship(source, value);
    }
}

/// <summary>
/// Returns the current relationship associated with the source, or null if
/// there is no relationship. Also sets a relationship between two objects. Null
/// can be passed as the property value, in which case the relationship will
/// be cleared.
/// </summary>
[SuppressMessage("Microsoft.Design", "CA1023:IndexersShouldNotBeMultidimensional")]
public MemberRelationship this[object sourceOwner, MemberDescriptor sourceMember]
{
    get
    {
        if (sourceOwner == null)
        {
            throw new ArgumentNullException(nameof(sourceOwner));
        }
        if (sourceMember == null)
        {
            throw new ArgumentNullException(nameof(sourceMember));
        }

        return GetRelationship(new MemberRelationship(sourceOwner, sourceMember));
    }
    set
    {
        if (sourceOwner == null)
        {
            throw new ArgumentNullException(nameof(sourceOwner));
        }
        if (sourceMember == null)
        {
            throw new ArgumentNullException(nameof(sourceMember));
        }

        SetRelationship(new MemberRelationship(sourceOwner, sourceMember), value);
    }
}
```

but the constructor doesn't allow nulls:
```cs
/// <summary>
/// This class represents a single relationship between an object and a member.
/// </summary>
public readonly struct MemberRelationship
{
    private readonly object _owner;
    private readonly MemberDescriptor _member;

    public static readonly MemberRelationship Empty = new MemberRelationship();

    /// <summary>
    /// Creates a new member relationship.
    /// </summary>
    public MemberRelationship(object owner, MemberDescriptor member)
    {
        _owner = owner ?? throw new ArgumentNullException(nameof(owner));
        _member = member ?? throw new ArgumentNullException(nameof(member));
    }
}
```

Reply:
Okay. I've undone this change - also lets us remove dead code as a result